perceval/backends/_slos.py

```diff
@@ -104,26 +104,28 @@ def compute(self, u, parent_coefs: Matrix = None, mk: int = None):
             child.compute(u, self.coefs, mk)
 
 
-class SLOSBackend(AProbAmpliBackend):
+class SLOSBackend(AStrongSimulationBackend):
     def __init__(self, mask=None, n=None, use_symbolic=False):
         super().__init__()
         self._reset()
         self._symb = use_symbolic
-        self._mask_str = mask
-        self._n = n
-        self._mask = None
+        if mask is not None:
+            self.set_mask(mask)
+        if n is not None:
+            get_logger().warn(
+                f"DeprecationWarning: 'n' parameter is now ignored and deprecated. version=0.12", channel.user)
```

Comment:
I refactored and found out the "n" info is not necessary. What's necessary is the photon count in the input state (and which had to be the exact "n" value, so that could be dangerous). After the refactoring, I'm ignoring the value, but still accepting the parameter (with the warning message) - so backward compatibility seems alright.